build-support/functions/20-build.sh

```diff
@@ -275,7 +274,28 @@ function build_consul {
       extra_dir="${extra_dir_name}/"
    fi
 
-   local container_id=$(docker create -it -e CGO_ENABLED=0 ${image_name} gox -os="${XC_OS}" -arch="${XC_ARCH}" -osarch="!darwin/arm !freebsd/arm !darwin/arm64" -ldflags "${GOLDFLAGS}" -output "pkg/bin/${extra_dir}{{.OS}}_{{.Arch}}/consul" -tags="${GOTAGS}")
+   # figure out if the compiler supports modules
+   local use_modules=0
+   if go help modules >/dev/null 2>&1
+   then
+      use_modules=1
+   elif test -n "${GO111MODULE}"
+   then
+      use_modules=1
+   fi
+
+   local volume_mount=
+   if is_set "${USE_MODULES}"
```

Comment:
should this be `use_modules`?

Reply:
yep

Reply:
I got local vs release mode mixed up and thought `make dev-docker` would activate this snippet when I initially ran it.

I wonder if it would be good to put a similar snippet over in the local build to do the `go_mod_assert` part during dev builds too.